api/src/routes/protected/chapter.test.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  devLogin,
+  setupServer,
+  superRequest,
+  createSuperRequest
+} from '../../../jest.utils';
+
+const MOCK_DATE = new Date('2024-11-08').getTime();
+
+describe('chapterRoutes', () => {
+  setupServer();
+
+  describe('Authenticated user', () => {
+    let setCookies: string[];
+    let superPost: ReturnType<typeof createSuperRequest>;
+
+    // Authenticate user
+    beforeAll(async () => {
+      setCookies = await devLogin();
+      superPost = createSuperRequest({ method: 'POST', setCookies });
+    });
+
+    describe('/chapter-completed', () => {
+      describe('validation', () => {
+        test('POST rejects requests without ids', async () => {
+          const response = await superPost('/chapter-completed');
+
+          expect(response.body).toStrictEqual({
+            type: 'error',
+            message: 'That does not appear to be a valid chapter submission.'
+          });
+          expect(response.statusCode).toBe(400);
+        });
+      });
+
+      describe('handling', () => {
+        beforeAll(() => {
+          jest.useFakeTimers({
+            doNotFake: ['nextTick']
+          });
+          jest.setSystemTime(MOCK_DATE);
+        });
+
+        afterAll(() => {
+          jest.useRealTimers();
+        });
```

Comment:
Using fake timers is way better than what I was messing around with! The only thing is that I would do differently is use a couple of dates and set the time during the test. I don't have a compelling reason, though, it's just that you'd expect the completion dates to differ.